pkg/service/replay/replay.go

```diff
@@ -186,16 +186,19 @@ func (r *Replayer) Start(ctx context.Context) error {
 	var testSetResult bool
 	testRunResult := true
 	abortTestRun := false
-
-	for i, testSetID := range testSetIDs {
-
-		testSetResult = false
-
+	var testSets []string
+	for _, testSetID := range testSetIDs {
 		if _, ok := r.config.Test.SelectedTests[testSetID]; !ok && len(r.config.Test.SelectedTests) != 0 {
 			continue
 		}
-
-		err := HookImpl.BeforeTestSetRun(ctx, testSetID)
+		testSets = append(testSets, testSetID)
```

Comment:
Lets sort the test-sets in alphabetical order, rest of the changes looks good to me.